core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1016,6 +1016,14 @@ package object config {
       .booleanConf
       .createWithDefault(true)
 
+  private[spark] val MAP_STATUS_COMPRESSION_CODEC =
+    ConfigBuilder("spark.shuffle.mapStatus.compression.codec")
```

Comment:
Let us mark it internal. Like the other code paths, we add such a conf. 

There does not exist a compression algorithm that can perform well for all the data patterns.

Reply:
+1 for @gatorsmile and @cloud-fan 's advice to make it `internal`.

Reply:
@Ngone51 . Could you update this PR?

Reply:
done b64e68d.